src/NuGet.Core/NuGet.ProjectModel/ProjectLockFile/PackagesLockFileUtilities.cs

```diff
@@ -464,5 +574,27 @@ public LockFileValidityWithMatchedResults(bool isValid, IReadOnlyList<KeyValuePa
             public static readonly LockFileValidityWithMatchedResults Invalid =
                 new LockFileValidityWithMatchedResults(isValid: false, matchedDependencies: null);
         }
+
+        /// <summary>
+        /// A class to return information about lock file validity with invalid reason.
+        /// </summary>
+        public class LockFileValidityWithInvalidReasons
```

Comment:
naming: How about `LockFileValididationResult`. 

The question I asked myself was what happens if you need to add another property in the future?
`LockFileValidityWithInvalidReasons(plusThisExtraStuff)`

Reply:
I am good with `LockFileValidationResult` class name. thank you.